services/statestorage/adapter/memory_persistence.go

```diff
@@ -4,41 +4,79 @@ import (
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/orbs-spec/types/go/protocol"
 	"github.com/pkg/errors"
-)
-
-type Config interface {
-}
+	)
 
 type InMemoryStatePersistence struct {
-	stateWritten chan bool
-	stateDiffs   map[primitives.ContractName]map[string]*protocol.StateRecord
-	config       Config
+	stateWritten 	chan bool
+	stateDiffs   	map[primitives.BlockHeight]map[primitives.ContractName]map[string]*protocol.StateRecord
 }
 
-func NewInMemoryStatePersistence(config Config) StatePersistence {
+func NewInMemoryStatePersistence() StatePersistence {
+	stateDiffsContract :=  map[primitives.ContractName]map[string]*protocol.StateRecord{primitives.ContractName("BenchmarkToken"): {}}
```

Comment:
Do we have to initialize "BenchmarkToken" by name here? If we avoid initializing it, won't it be initialized on first write? or do you want to avoid that it fails on first read?

Reply:
we were told that the provider will create the contract so that we have a default by name for now.